Gekko/UI/MainViewController.swift

```diff
@@ -126,26 +127,54 @@ class MainViewController : UIViewController,
 
         ordersDataFacade = CoreDataFacade(completionBlock: { [weak self] in
             DispatchQueue.main.async { [weak self] in
-                self?.scheduleOrdersStatusUpdating{}
+                self?.scheduleOrdersStatusUpdating()
             }
         })
     }
 
-    fileprivate func scheduleOrdersStatusUpdating(onCompletion: @escaping () -> Void) {
-        updateOrdersStatus(forCurrencyPair:.BtcUah)
-        updateOrdersStatus(forCurrencyPair:.EthUah)
-        updateOrdersStatus(forCurrencyPair:.LtcUah)
+    fileprivate func scheduleOrdersStatusUpdating() {
+        handleOrdersStatusUpdating(onCompletion: {})
 
         DispatchQueue.main.asyncAfter(deadline:DispatchTime.now() + MainViewController.OrdersPollTimeout) {
             [weak self] () in
             if (self != nil) {
-                self!.scheduleOrdersStatusUpdating{}
+                self!.scheduleOrdersStatusUpdating()
             }
         }
     }
 
-    fileprivate func updateOrdersStatus(forCurrencyPair currencyPair:BTCTradeUACurrencyPair) {
+    fileprivate func handleOrdersStatusUpdating(onCompletion:@escaping () -> Void) {
+        let RequiredOperationsCount = 3
+        var operationsCount = 0
+        let completionHandler = {
+            operationsCount += 1
+        
+            if operationsCount == RequiredOperationsCount {
+                onCompletion()
+            }
+        }
+        
+        updateOrdersStatus(forCurrencyPair:.BtcUah, onCompletion:completionHandler)
+        updateOrdersStatus(forCurrencyPair:.EthUah, onCompletion:completionHandler)
+        updateOrdersStatus(forCurrencyPair:.LtcUah, onCompletion:completionHandler)
+    }
+    
+    fileprivate func updateOrdersStatus(forCurrencyPair currencyPair:BTCTradeUACurrencyPair,
+                                        onCompletion:@escaping () -> Void) {
         if let orders = ordersDataFacade?.orders(forCurrencyPair:currencyPair.rawValue as String) {
+            let RequiredOrdersCount = orders.count
```

Comment:
requiredOrdersCount (This is not literal, it's calculated variable which is not modified further)